cmd/config/internal/commands/cmdcreatesetter.go

```diff
@@ -184,17 +183,7 @@ func (r *CreateSetterRunner) set(c *cobra.Command, args []string) error {
 		if err != nil {
 			return err
 		}
-
-		r.CreateSetter.OpenAPIFileName = openAPIFileName
-		resourcePackagesPaths, err := pathutil.DirsWithFile(args[0], openAPIFileName, r.CreateSetter.RecurseSubPackages)
-		if err != nil {
-			return err
-		}
-
-		if len(resourcePackagesPaths) == 0 {
-			return errors.Errorf("unable to find %q in package %q", r.CreateSetter.OpenAPIFileName, args[0])
-		}
-		for _, resourcesPath := range resourcePackagesPaths {
+		executeCmd := func(pkgPath string) error {
```

Comment:
Large inline functions like this is a bit hard to read. I think we should try to implement this as a separate function (that might belong to the `CreateSetterRunner` in this case) and then pass that into the call to `executeCmdOnPkgs`. 

Similarly, I think it would be cleaner if the `executeCmdOnPkgs` accept a non-function interface type as the parameter. We can make this easy to use by having a function implement the interface. There is an example of this pattern here: https://github.com/kubernetes-sigs/kustomize/blob/e7970d82a8eea1e1eb917b80506f664d51ccedb2/api/filters/imagetag/imagetag.go#L31

WDYT?

Reply:
Done. Passed an interface.